### ISSUE TITLE
Incorrect update window UI

### ISSUE THREAD

**Issue by Flynsarmy** [ramokz/phantom-camera#260]
### Issue description

Update window has no 'Yes' button visible. I have a 4k monitor with Editor - Display Scale set to 175% and the phantom camera update UI not only has very small text that's hard to read, but also I have no idea what the 'No' button does or how it differs from the 'Close' button, and I assume there's supposed to be a 'Yes' or 'Update' button somewhere that isn't visible.
![image](https://github.com/ramokz/phantom-camera/assets/334808/cea7970d-419d-450d-8381-40c2450680e9)

### Steps to reproduce

Wait for an update. It'll pop up in the center of the screen on starting Godot.

**Comment by tayete** [comment]
Same here.

**Comment by tayete** [comment]
When clicking "No" a dropdown menu shows also the "Yes" option, thus it seems to work well, but isn't clear at first how to proceed with the popup update window.

**Comment by ramokz** [comment]
Don't have a high-resolution display, so can't debug or fix this easily unfortunately, but my understanding is that UI nodes, at least with Godot editor tools, require some manual intervention when scaling is being applied.

Aside from the text size, the only real issue I can see is that the close button is sitting too close to the dropdown, which relates to the issue mentioned above.

@Flynsarmy does pressing the `No` button not render a dropdown menu?

**Comment by tiloc** [comment]
I ran into the same issue with a stock installation of macOS and Godot 4.2.2 (no special scaling settings) on a 4k monitor. I ended up manually deleting the plugin from the filesystem and downloading manually from Github. I do not recall a dropdown when I hit No, but my memory is already a bit hazy.

**Comment by audeck** [comment]
The original problem is simply that the `OptionButton` isn't styled as a dropdown, but rather more so as a button with an action ("No" as in "No, I don't want this update."), which, in my opinion, is completely valid.

A solution would be to either change the styling (to include a downward facing chevron, etc.) in order to make the fact the button is a dropdown more clear, or swap the `OptionButton` for a `CheckBox` or `CheckButton`.

If you want the users to be _really_ sure they know what they're doing, you can add multiple check boxes in multiple stages (i.e. once they check the first one, another one appears with a text akin to "Are you REALLY sure you know what you're doing?"). The download button would show up only after checking the last one.

**Comment by Flynsarmy** [comment]
> @Flynsarmy does pressing the `No` button not render a dropdown menu?

I thought it was a button and not a drop down so I hit close, manually removed the add-on and downloaded the new one from GitHub.



> I think the most common UI seen in other installers would be to have 2 buttons, one with "Skip this version" and one with "Download and Install".

I agree with this. Very clear what your options are.

**Comment by ramokz** [comment]
> I think the most common UI seen in other installers would be to have 2 buttons, one with "Skip this version" and one with "Download and Install".

That is what normally appears whenever smaller updates are released — can see that version on the [documentation page](https://phantom-camera.dev/editor-updater). However, for larger updates this is very much intentionally made not to be like that.

As the red text in that updater variant also suggests, it basically comes down to adding a deliberate interaction friction so that people don't immediately click “Update” and potentially break their current project. The added friction is meant to give a moment of pause and require a bit more thought before proceeding. It's worth adding that because that updater variant only appears for larger releases, it is fairly uncommon to see it, which is another reason why there's a fairly stark interaction difference.

> The original problem is simply that the OptionButton isn't styled as a dropdown, but rather more so as a button with an action ("No" as in "No, I don't want this update."), which, in my opinion, is completely valid.
> A solution would be to either change the styling (to include a downward facing chevron, etc.) in order to make the fact the button is a dropdown more clear, or swap the OptionButton for a CheckBox or CheckButton.

The assumption I'm hearing is that the updater doesn't, at a glance, have a way to update to the release it promotes, i.e. it's bugged or only functions as a notification without steps for what to do next. Do think there's validity in thinking that the button says, “No, thanks” and so adding a chevron might alleviate that confusion. It shouldn't be difficult to add, but like mentioned above, it's a rare updater variant, so it's obviously not a priority.

As for the idea for a checkbox  / checkbutton, I did consider that initially, but decided to go for a dropdown menu in the end because it introduced higher friction, which, again, was the intent.

> If you want the users to be really sure they know what they're doing, you can add multiple check boxes in multiple stages (i.e. once they check the first one, another one appears with a text akin to "Are you REALLY sure you know what you're doing?"). The download button would show up only after checking the last one.

Having multiple checkboxes is a valid alternative that I hadn't considered at the time. Conversely, my fear with that approach is that people will just toggle the checkboxes quickly and press update, as that's kinda what most do for all terms of service and cookie consent forms given their omnipotent presence today.
